pkg/scheduler/actions/common/allocate.go

```diff
@@ -27,6 +27,22 @@ func AllocateJob(ssn *framework.Session, stmt *framework.Statement, nodes []*nod
 		return false
 	}
 
+	nodePartitions, err := ssn.NodePartitionFn(job, tasksToAllocate, nodes)
+	if err != nil {
+		log.InfraLogger.Errorf("Failed to partition nodes for job <%s/%s> in session %v, err: %v", job.Namespace, job.Name, ssn.UID, err)
+		return false
+	}
+
+	for _, partition := range nodePartitions {
```

Comment:
Please add a high-level log (maybe 5 or 6) that will give us details on the partitions (how many, attempt x/N, names, etc)